pytorch_blade/tests/disc/testing_base.py

```diff
@@ -42,27 +44,41 @@ def _ScriptFunction2Module(self, nn_module):
         else:
             return nn_module
 
-    def cvt_to_disc(self, nn_module, test_data):
+    def cvt_to_disc(self, nn_module, test_data, input_dims=[]):
         cfg = Config.get_current_context_or_new()
         cfg.optimization_pipeline = mlir.backend_name()
+        cfg.annotate_args = input_dims
         with mlir.testing_context(), support_fusion_group.min_group_nodes(1), cfg:
             nn_module = self._ScriptFunction2Module(nn_module)
             nn_module = nn_module.eval().to(self.device)
             opt_module = optimize(
                 nn_module,
                 allow_tracing=True,
-                model_inputs=test_data,
+                model_inputs=test_data
             )
         return opt_module
 
     def _test_cvt_to_disc(
-        self, nn_module, test_data, rtol=1e-6, atol=1e-3, n_engines=1
+        self, nn_module, test_data, dims=[], rtol=1e-6, atol=1e-3, n_engines=1
     ):
         nn_module = self._ScriptFunction2Module(nn_module)
         nn_module = nn_module.eval().to(self.device)
         result = nn_module(*test_data)
-        opt_module = self.cvt_to_disc(nn_module, test_data)
+        opt_module = self.cvt_to_disc(nn_module, test_data, dims)
         output = opt_module.forward(*test_data)
         self.assertEqual(output, result, rtol=rtol, atol=atol)
         self.assertGreaterEqual(mlir.num_engines(opt_module), n_engines)
         return output, result
+
+    def _gen_test_data(self, dims):
+        test_data = []
+        for dim in dims:
+            for i, dim_i in enumerate(dim):
+                if dim_i == -1:
+                    dim[i] = random.randint(1, 100)
```

Comment:
Should the seed the fixed, so that we can re-produce tests?

Reply:
done.